### PR TITLE
New type of task (smtp) to send emails using an SMTP server

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -45,12 +45,34 @@ mongo-dbname: workflows
 
 # mongo_password: secret
 
+# SMTP host
```
Comment (alfrunes) [inline config.yaml]: Is there any reason why these parameters aren't included in the (`SMTPTask`) task definition?

File: config.yaml [hunk 2]
```diff
@@ -45,12 +45,34 @@ mongo-dbname: workflows
...
+# Defaults to: localhost:25
+# Overwrite with environment variable: WORKFLOWS_SMTP_HOST
+
+# smtp_host: "localhost:25"
+
+# SMTP username
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply config.yaml]: Sure, I'm adding it

File: config.yaml (hunk 1 above)
Comment (tranchitella) [reply config.yaml]: My idea is that smtp host/username/password are deployment settings, which you can easily override using env variables both in docker-compose and k8s. If we put those configuration in the definition of the tasks, they'll be built in the docker image making it impossible to change them at deploy time.

Am I wrong?

Comment (alfrunes) [reply config.yaml]: You're probably right. I was just wondering if we're tailoring the task definition a little too much to our current setup/configs. However, I don't know if there will ever be use for configuring workflows with different smtp-hosts/e-mail accounts, but I just thought I'd to bring it up anyway.

Comment (tranchitella) [reply config.yaml]: I believe it is not a wise choice to add deployment-specific settings to workflow definitions (SMTP settings, in this case).  I don't think we are going to use multiple SMTPs in the near future, so better to leave it as it is (SMTP settings from config/env variables) and manage it differently in the future when such a need arises.

Thanks for your review, anyway!

**Pull request merged**